caos.ansible_roles/roles/install_crowdstrike_falcon/vars/main.yaml

```diff
@@ -26,95 +26,215 @@ mac_hash: "cef4f59bc87e35fabe490e0f34b2f4dfaa076822e5a61d0a5fffc58014d446a1"
   - hostname: "arm64:redhat-7.6"
     hash: "{{ al2_arm64_hash }}"
+    distribution: ""
+    major_version: ""
+    architecture: ""
   - hostname: "amd64:al-2"
     hash: "{{ al2_amd64_hash }}"
+    distribution: "Amazon"
+    major_version: "2"
+    architecture: "x86_64"
   - hostname: "arm64:al-2"
     hash: "{{ al2_arm64_hash }}"
+    distribution: "Amazon"
+    major_version: "2"
+    architecture: "aarch64"
 
 apt:
   - hostname: "amd64:ubuntu22.04"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Ubuntu"
+    major_version: "22"
+    architecture: "x86_64"
   - hostname: "amd64:ubuntu20.04"
     hash: "{{ ubuntu_debian_amd64_hash}}"
-  - hostname: "amd64:ubuntu18.04" 
+    distribution: "Ubuntu"
+    major_version: "20"
+    architecture: "x86_64"
+  - hostname: "amd64:ubuntu18.04"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Ubuntu"
+    major_version: "18"
+    architecture: "x86_64"
   - hostname: "amd64:ubuntu16.04"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Ubuntu"
+    major_version: "16"
+    architecture: "x86_64"
   - hostname: "arm64:ubuntu22.04"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Ubuntu"
+    major_version: "22"
+    architecture: "aarch64"
   - hostname: "arm64:ubuntu20.04"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Ubuntu"
+    major_version: "20"
+    architecture: "aarch64"
   - hostname: "arm64:ubuntu18.04"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Ubuntu"
+    major_version: "18"
+    architecture: "aarch64"
   - hostname: "arm64:ubuntu16.04"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Ubuntu"
+    major_version: "16"
+    architecture: "aarch64"
   - hostname: "amd64:debian-buster"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Debian"
+    major_version: "10"
+    architecture: "x86_64"
   - hostname: "amd64:debian-bullseye"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Debian"
+    major_version: "11"
+    architecture: "x86_64"
   - hostname: "amd64:debian-bookworm"
     hash: "{{ ubuntu_debian_amd64_hash}}"
+    distribution: "Debian"
+    major_version: "12"
+    architecture: "x86_64"
     # Debian arm64 not supported by CrowdStrike, using Ubuntu arm64 installers instead
   - hostname: "arm64:debian-buster"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Debian"
+    major_version: "10"
+    architecture: "aarch64"
   - hostname: "arm64:debian-bullseye"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Debian"
+    major_version: "11"
+    architecture: "aarch64"
   - hostname: "arm64:debian-bookworm"
     hash: "{{ ubuntu_arm64_hash }}"
+    distribution: "Debian"
+    major_version: "12"
+    architecture: "aarch64"
 
 zyp:
   - hostname: "amd64:sles-12.3"
     hash: "{{ sles12_amd64_hash }}"
+    distribution: "SLES_SAP"
+    major_version: "12"
+    architecture: "x86_64"
```

Comment:
Hmmm unfortunately this is the one we're using for our tests (SLES distro with SAP installed). Probably CAOS could tell us what AMIs they are using, in case we can unify the images used for the testing, as I suspect these are quite Logging-specific. 🤔

Reply:
Even if we use specific AMIs, this should not be a problem: we can just add them as separate items, so they are not tied to a particular hostname. In fact, thinking it better, I'd suggest to adopt this way instead:

```
IGNORE_HOSTNAME: "IGNORED"
IGNORE_ANSIBLE_FACTS: "IGNORED"

...
  zyp:
    - hostname: "amd64:sles-12.3"
      hash: "{{ sles12_amd64_hash }}"
      distribution: "{{ IGNORE_ANSIBLE_FACTS }}"
      major_version: "{{ IGNORE_ANSIBLE_FACTS }}"
      architecture: "{{ IGNORE_ANSIBLE_FACTS }}"
     - hostname: "{{ IGNORE_HOSTNAME }}"
      hash: "{{ sles12_amd64_hash }}"
      distribution: "SLES_SAP"
      major_version: "12"
      architecture: "x86_64"
```

Or even better, can we just allow two different schemas and allow empty values in the list objects?:
```
  zyp:
    - hostname: "amd64:sles-12.3"
      hash: "{{ sles12_amd64_hash }}"
     - distribution: "SLES_SAP"
      major_version: "12"
      architecture: "x86_64"
      hash: "{{ sles12_amd64_hash }}"
```

This way, if someone else needs to support another SLES12 distribution, they can just add it to the list.

Reply:
That is a good point since there are no raw sles15/12 specific SP distros available apparently. I agree that if CAOS is using a different distribution we should switch to it too.